### PR TITLE
Resolve duplicate cypresses

### PR DESCRIPTION
**Pull request by rserur**
# Fullstack PR Summary

[Pivotal Ticket](https://www.pivotaltracker.com/story/show/180576998)

## Included Changes:

- Remove duplicate cypress directory in `spec/` -- reinstall at root (using cypress-on-rails commands)
- Add rails s to test:e2e script and headless version
- Remove Post-related example specs

## Notes to Reviewer
Since cypress_on_rails was initially configured to look at the `/spec/cypress` directory, any specs placed in `cypress` probably didn't work quite right or at all.

## Checklist
- [x] Have you updated documentation (if needed)?
- [x] Have you moved the ticket to "In Review"?

### CODE REVIEW

File: cypress/support/on-rails.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ Cypress.Commands.add("appCommands", function (body) {
       failOnStatusCode: false,
     })
     .then((response) => {
+      log.end();
```
Comment (sequelster) [inline cypress/support/on-rails.js:16]: ?

File: package.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
     "test:js": "jest",
     "test:ruby": "bundle exec rake test spec",
     "test:watch": "jest --watch",
-    "test:e2e": "cypress open",
+    "test:e2e": "rails s -e test & cypress open",
```
Comment (sequelster) [inline package.json:18]: So cypress open still usually sticks a foreground process of running logs in my console and given rails s here is also foreground...are we missing logs from one or the other when run this way? (I normally run them separately and totally forget to open the server sometimes, so I do understand and appreciate the handiness of running at the same time :D)

**Review by sequelster — COMMENTED** [review]
I had originally gotten rid of the spec examples, but I think Lael wanted them kept around for reference. We can re-evaluate that. I think we have enough real tests where maybe it's not necessary anymore? but idk the app commands might still be useful to keep around.

File: config/initializers/cypress_on_rails.rb [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
+    c.cypress_folder = File.expand_path("#{__dir__}/../../cypress")
     # WARNING!! CypressOnRails can execute arbitrary ruby code
     # please use with extra caution if enabling on hosted servers or starting your local server on 0.0.0.0
     c.use_middleware = !Rails.env.production?
     c.logger = Rails.logger
   end
 
   # if you compile your assets on CI
-  if ENV["CYPRESS"].present?
+  if ENV["CYPRESS"].present? && ENV["CI"].present?
```
[the remark this answers is not in the document]
Comment (sequelster) [reply config/initializers/cypress_on_rails.rb]: ^ No that is exactly why I did. Either we use ENV["CI"] (which is built-in to the github action environment) or we keep ENV["CYPRESS"], both are not actually needed.

**Review by raq929 — APPROVED** [review]
LGTM!

File: package.json (hunk 2 above)
Comment (raq929) [reply package.json:18]: This makes sure it's running in the test environment, so you don't accidentally wipe all your test data. Ask me how I know. 😛

File: cypress/support/on-rails.js (hunk 1 above)
Comment (rserur) [reply cypress/support/on-rails.js:16]: Reinstalling cypress on rails added this 🤷‍♀️

File: cypress/app_commands/scenarios/basic.rb [hunk 4]
```diff
@@ -1,3 +1,5 @@
 # frozen_string_literal: true
 
-Cat.create!(name: "Athena", breed: "Siamese", age: 4, personality: "Confused")
+# You can setup your Rails state here
+# MyModel.create name: 'something'
+Cat.create(name: "Athena", breed: "Siamese", age: 4, personality: "Confused")
```
[the remark this answers is not in the document]
Comment (rserur) [reply cypress/app_commands/scenarios/basic.rb:5]: Ah, didn't mean to get rid of this-- it raises an error upon failure, which I think is smart here-- I'll restore it.

File: config/initializers/cypress_on_rails.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (rserur) [reply config/initializers/cypress_on_rails.rb]: Reinstalling cypress on rails added this-- Talia confirmed it's not needed so will remove

File: package.json (hunk 2 above)
Comment (rserur) [reply package.json:18]: That's a good point about helpful logs, though-- I'll see if I can add [Concurrently](https://yarnpkg.com/package/concurrently), which helps print out outputs from concurrent scripts more clearly.

Comment (rserur) [reply package.json:18]: Looks like concurrently works fine, so that'll address this:
image

Comment (sequelster) [reply package.json]: oooo that's cool!

**Pull request merged**